From fc91be029353e7f7743c938863e2ee69c507b275 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Fri, 12 Feb 2021 17:04:16 +0200
Subject: [PATCH] [SW-20154] make: compile libraries based on device

libraries like sbtool and ellipsys are needed only
for Goya2. hence limit their compilation only to
that device.

Change-Id: Id6743058b502bc45fc13633cf718ed102073bbaf
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/Makefile b/Makefile
index 2cade4d..12283c7 100644
--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@
 
 ifneq ($(CROSS_COMPILE),)
 LDFLAGS = -flto
-LDFLAGS += -L $(LIBDIR) -lsb -lellipsys -L $(HL_DEV)
+LDFLAGS += -L $(LIBDIR) -L $(HL_DEV)
 LDFLAGS += -T $(HL_DEV)/$(HL_TARGET)_$@.lds
 LDFLAGS += -nostartfiles
 LDFLAGS += -Wl,-e__start
@@ -54,15 +54,11 @@
 ARCBOOT_ROM_OBJECTS := $(ARCBOOT_COMMON_SRCS_C:.c=.o) $(ARCBOOT_ROM_SRCS_C:.c=.o) $(ARCBOOT_ROM_SRCS_S:.S=.o)
 endif
 
-.PHONY: all libsb libellipsys sbtool
-all: libsb libellipsys
+.PHONY: all
+all:
 	@$(MAKE) arcboot_rom.bin
 	@$(MAKE) arcboot_ram.bin
 
-libsb libellipsys:
-	@+$(MAKE) -C sdk $@ SB_SIGN_MODE=rsa_fixed > /dev/null
-sbtool:
-	@$(MAKE) -C sdk $@ > /dev/null
 %.bin: %
 	@$(OBJCOPY) -O binary -R .note.gnu.build-id -R .test* $< $@
 
diff --git a/goya2_arc/Makefile b/goya2_arc/Makefile
index e69de29..9e1a702 100644
--- a/goya2_arc/Makefile
+++ b/goya2_arc/Makefile
@@ -0,0 +1,9 @@
+LDFLAGS += -lsb -lellipsys
+
+.PHONY: libsb libellipsys sbtool
+all: libsb libellipsys
+
+libsb libellipsys:
+	@+$(MAKE) -C sdk $@ SB_SIGN_MODE=rsa_fixed > /dev/null
+sbtool:
+	@$(MAKE) -C sdk $@ > /dev/null
